src/Emit.hs

```diff
@@ -313,8 +312,6 @@ toC toCMode (Binder meta root) = emitterSrc (execState (visit startingIndent roo
                     -- This requires a bunch of extra machinery though, so this will do for now...
                     [var ++ periodOrArrow ++ "_tag == " ++ tagName caseTy (removeSuffix caseName)] ++
                       concat (zipWith (\c i -> tagCondition (var ++ periodOrArrow ++ "u." ++ removeSuffix caseName ++ ".member" ++ show i) "." (forceTy c) c) caseMatchers ([0..] :: [Int]))
-                      where notUnitX (XObj _ _ (Just UnitTy)) = False
```

Comment:
Ah, this is possibly a bug--probably we want to filter Units from `caseMatchers` here. Mind adding a todo to double check this?

Reply:
Done. Would it need `isUnit` or the one removed that just checks plain units?

Reply:
iirc the list was a list of xobjs in this case, so `isUnit` will work but will need to be composed w/ `forceTy`, I think `filter (isUnit . forceTy)` should do it